src/superphot_plus/classify_ztf.py

```diff
@@ -23,180 +30,321 @@
-    num_layers,
-    classify_log_file,
-    include_redshift=False,
-    num_folds=NUM_FOLDS,
-    fits_plotted=False,
-    metrics_dir=METRICS_DIR,
-    models_dir=MODELS_DIR,
-    csv_path=None,
-    cm_folder=CM_FOLDER,
-    sampler="dynesty",
-):
-    """Train MLP to classify between supernovae of 'allowed_types'.
-
     Parameters
     ----------
-    goal_per_class : int
-        Oversampling such that there are this many fits per supernova
-        type.
-    num_epochs : int
-        Number of training epochs.
-    neurons_per_layer : int
-        Number of neurons per hidden layer of MLP.
     num_layers : int
         Number of hidden layers in MLP.
-    fits_plotted : bool
-        If true, assumes all sample fit plots are saved in
-        FIT_PLOTS_FOLDER. Copies plots of wrongly classified samples to
-        separate folder for manual followup. Defaults to False.
+    neurons_per_layer : int
+        Number of neurons per hidden layer of MLP.
+    goal_per_class : int
+        Oversampling such that there are this many fits per supernova type.
+    metrics_dir : str
+        The directory where model metrics are stored.
+    models_dir: str
+        The directory where trained models are stored.
+    cm_folder : str
+        The directory where plots of confusion matrices are stored.
+    classify_log_file : str
+        The output file where classifications are logged.
+    sampler : str
+        The type of sampler used for the lightcurve fits.
+    include_redshift : bool
+        If True, includes redshift data for training.
     """
 
-    os.makedirs(metrics_dir, exist_ok=True)
-    os.makedirs(models_dir, exist_ok=True)
-    os.makedirs(cm_folder, exist_ok=True)
+    def __init__(
+            self,
+            num_layers,
+            neurons_per_layer,
+            goal_per_class,
+            metrics_dir=METRICS_DIR,
+            models_dir=MODELS_DIR,
+            cm_folder=CM_FOLDER,
+            classify_log_file=CLASSIFY_LOG_FILE,
+            sampler="dynesty",
+            include_redshift=True,
+    ):
+        self.num_layers = num_layers
+        self.neurons_per_layer = neurons_per_layer
+        self.goal_per_class = goal_per_class
+        self.sampler = sampler
+        self.include_redshift = include_redshift
+
+        self.metrics_dir = metrics_dir
+        self.models_dir = models_dir
+        self.cm_folder = cm_folder
+        self.classify_log_file = classify_log_file
+
+        # Derive from sampler type
+        self.fits_dir = f"{DATA_DIR}/{sampler}_fits"
+
+        # Initialize output directories
+        os.makedirs(metrics_dir, exist_ok=True)
+        os.makedirs(models_dir, exist_ok=True)
+        os.makedirs(cm_folder, exist_ok=True)
+        os.makedirs(FIT_PLOTS_FOLDER, exist_ok=True)
+
+        # Cleanup previous output content
+        for directory in [metrics_dir, models_dir, cm_folder, FIT_PLOTS_FOLDER]:
+            files = glob.glob(os.path.join(directory, "*"))
```

Comment:
Maybe swap these operations? You could use `shutil.rmtree` to remove the directory and all contents recursively, then recreate the directory.